internal/stacks/stackruntime/internal/stackeval/component.go

```diff
@@ -209,6 +198,12 @@ func (c *Component) ResultValue(ctx context.Context, phase EvalPhase) cty.Value
 			return cty.DynamicVal
 		}
 
+		if insts[addrs.WildcardKey] != nil {
```

Comment:
We should still have a value for this instead of DynamicValue. The plan should still have executed with the ExternalDeferral field set to true and we should have got anything back that the plan could still have computed, even if that is completely unknown.

We still want to execute the plan, even if the insts are unknown, just with everything in the plan being deferred.